src/Language/PureScript/Interactive/Parser.hs

```diff
@@ -79,38 +103,44 @@ parseDirective cmd =
     Reload   -> return ReloadState
     Clear    -> return ClearState
     Paste    -> return PasteLines
-    Browse   -> BrowseModule <$> parseRest P.moduleName arg
+    Browse   -> BrowseModule . CST.nameValue <$> parseRest (parseOne CST.parseModuleNameP) arg
     Show     -> ShowInfo <$> parseReplQuery' arg
-    Type     -> TypeOf <$> parseRest P.parseValue arg
-    Kind     -> KindOf <$> parseRest P.parseType arg
+    Type     -> TypeOf . CST.convertExpr "" <$> parseRest (parseOne CST.parseExprP) arg
+    Kind     -> KindOf . CST.convertType "" <$> parseRest (parseOne CST.parseTypeP) arg
     Complete -> return (CompleteStr arg)
-    Print    -> parseRest
-                  ((eof *> return (ShowInfo QueryPrint))
-                  <|> (SetInteractivePrint <$> parseFullyQualifiedIdent))
-                  arg
+    Print
+      | arg == "" -> return $ ShowInfo QueryPrint
+      | otherwise -> SetInteractivePrint <$> parseRest (parseOne parseFullyQualifiedIdent) arg
 
 -- |
 -- Parses expressions entered at the PSCI repl.
 --
-psciExpression :: P.TokenParser Command
-psciExpression = Expression <$> P.parseValue
+psciExpression :: CST.Parser Command
+psciExpression = Expression . CST.convertExpr "" <$> CST.parseExprP
 
 -- | Imports must be handled separately from other declarations, so that
 -- :show import works, for example.
-psciImport :: P.TokenParser Command
-psciImport = do
-  (mn, declType, asQ) <- P.parseImportDeclaration'
-  return $ Import (mn, declType, asQ)
+psciImport :: FilePath -> CST.Parser Command
+psciImport filePath = do
+  decl <- CST.convertImportDecl filePath <$> CST.parseImportDeclP
+  case decl of
+    P.ImportDeclaration _ mn declType asQ ->
+      pure $ Import (mn, declType, asQ)
+    _ -> error "Not an import"
```

Comment:
should we use `internalError` here?

Reply:
Probably. It would be very very bad and everything would break if the error case was possible. The only reason it's there is because there isn't a `convert` function for just the import tuple. I could also probably just add that instead and remove the error.